### ISSUE TITLE
How to do not download image judge by the image's size?

### ISSUE THREAD

**Issue by liaodalin19903** [rs/SDWebImage#1768]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios 
 Platform Version        | e.g. 8.0
 SDWebImage Version      | 4.0.0-beta2
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

If the image's size is very large, I use :
    
    [_iconView sd_setImageWithURL:[NSURL URLWithString:model.lml_iconName] placeholderImage:img_placeholder_circle_header];

Sometims my app will crash.
 So, how to judge the image's size if the size is in limit, I can `setImageWithURL`, if not I do not set the `setImageWithURL`. Because the method will download image, the image is too large, it can not download, or app will crash.

**Comment by RunesReader** [comment]
You can try this
 `[image sd_setImageWithURL:avatarPath  placeholderImage:nil options:SDWebImageScaleDownLargeImages];`
Flag SDWebImageScaleDownLargeImages can help you.
